services-js/registry-certs/client/birth/QuestionsPage.tsx

```diff
@@ -0,0 +1,158 @@
+
+interface Props extends Pick<PageDependencies, 'birthCertificateRequest'> {}
+
+/**
+ * Guides the user through a number of questions, step by step, in order to
+ * provide the Registry with the information they will need to locate the
+ * birth record.
+ *
+ * User will progress to /review upon completion of this workflow.
+ */
+@observer
+export default class QuestionsPage extends React.Component<Props> {
+  private answerQuestion = (
+    question: Step,
+    answers: Partial<BirthCertificateRequestInformation>
+  ): void => {
+    const {
+      answerQuestion,
+      setCurrentStep,
+      steps,
+    } = this.props.birthCertificateRequest;
+    const currentIndex = steps.indexOf(question);
+
+    // We need to convert this value to a boolean for the request state object.
+    if (question === 'forWhom') {
+      answerQuestion({
+        ...answers,
+        forSelf: answers.forSelf === ('true' as any),
+      });
+    } else {
+      answerQuestion(answers);
+    }
+
+    // If ID verification is not known to be necessary, advance to the
+    // review page.
+    // Otherwise, continue to verifyIdentification step. When completed,
+    // then advance to the review page.
+    if (
+      question === 'parentalInformation' &&
+      !steps.includes('verifyIdentification')
+    ) {
+      Router.push('/birth/review');
+    } else if (question === 'verifyIdentification') {
+      Router.push('/birth/review');
+    } else {
+      setCurrentStep(steps[currentIndex + 1]);
+    }
+  };
+
+  private stepBackOneQuestion = (): void => {
+    const {
+      steps,
+      currentStep,
+      setCurrentStep,
+    } = this.props.birthCertificateRequest;
+    const currentIndex = steps.indexOf(currentStep);
+
+    // Ensure we cannot go back any further than the first question.
+    const newIndex = currentIndex > 0 ? currentIndex - 1 : 0;
+
+    setCurrentStep(steps[newIndex]);
+  };
+
+  // Clear all data and return to initial question.
+  private handleUserReset = (): void => {
+    this.props.birthCertificateRequest.clearBirthCertificateRequest();
+  };
+
+  public render() {
+    const answers = this.props.birthCertificateRequest.requestInformation;
+    const {
+      steps,
+      currentStep,
+      currentStepCompleted,
+      setCurrentStepCompleted,
+    } = this.props.birthCertificateRequest;
+
+    return (
+      <PageWrapper
+        totalSteps={steps.length}
+        currentStep={steps.indexOf(currentStep) + 1}
+        currentStepCompleted={currentStepCompleted}
+        showProgress={true}
+      >
+        <Head>
+          <title>Boston.gov — Request a Birth Certificate</title>
+        </Head>
+
+        {currentStep === 'forWhom' && (
+          <ForWhom
+            forSelf={answers.forSelf}
+            howRelated={answers.howRelated}
+            handleStepCompletion={setCurrentStepCompleted}
+            handleProceed={a => this.answerQuestion('forWhom', a as any)}
```

Comment:
The `forWhom` question is a weird case, because the value is handled as a string inside the component (because Formik), but is a boolean in the request state.